### PR TITLE
SONAR-7180 remove properties for deleted component page from DB

### PR DESCRIPTION
**Pull request by sns-seb**
uses migration id 1009, must be merged after #710

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Protected method 'configureModule(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/de1436150361a4be86c788e81f1f97a96e0abbf4/sonar-db/src/main/java/org/sonar/db/version/MigrationStepModule.java#L68) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+#
+# SonarQube 5.2
+# SONAR-6044
```
Comment (julienlancelot) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb]: Correct ticket is SONAR-7180

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+#
+# SonarQube 5.2
```
Comment (julienlancelot) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb]: Correct version is 5.4

File: sonar-db/src/main/java/org/sonar/db/version/v54/RemoveComponentPageProperties.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Remove properties use to store customisation of dropped component page:
+ * <ul>
+ *   <li>sonar.core.projectsdashboard.columns</li>
+ *   <li>sonar.core.projectsdashboard.defaultSortedColumn</li>
+ *   <li>any other property starting with "sonar.core.projectsdashboard."</li>
+ * </ul>
+ */
+public class RemoveComponentPageProperties extends BaseDataChange {
```
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/version/v54/RemoveComponentPageProperties.java]: Unit tests are missing !

**Comment by julienlancelot** [comment]
@seb unit tests are missing

File: sonar-db/src/main/java/org/sonar/db/version/v54/RemoveComponentPageProperties.java (hunk 3 above)
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/version/v54/RemoveComponentPageProperties.java]: I so don't want to add yet another slow DB test using H2...

Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/version/v54/RemoveComponentPageProperties.java]: It's the only one to have a proper unit test.

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb (hunk 1 above)
Comment (sns-seb) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb]: thanks

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb (hunk 2 above)
Comment (sns-seb) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1009_remove_component_page_properties.rb]: well spotted

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Protected method 'configureModule(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/bc5811ac6518df8d263430a7a7085e6f88f762d3/sonar-db/src/main/java/org/sonar/db/version/MigrationStepModule.java#L68) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

**Comment by sns-seb** [comment]
unit tests added and feedback applied, back to you @julienlancelot

File: sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    assertThat(db.countRowsOfTable(TABLE_PROPERTIES)).isEqualTo(2);
+  }
+
+  private void assertPropertiesContainsRow(String key, @Nullable Integer resourceId, @Nullable Integer userId) {
+    assertThat(db.countSql(propertiesRowSql(key, resourceId, userId))).isEqualTo(1);
+  }
+
+  private static String propertiesRowSql(String key, @Nullable Integer resourceId, @Nullable Integer userId) {
+    return format(
+        "select count(*) from properties where prop_key='%s' and resource_id %s and user_id %s and text_value = '%s'",
```
Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java]: indentation seems bad here

File: sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  private static String whereClauseOfInteger(@Nullable Integer id) {
+    if (id == null) {
+      return "is null";
+    }
+    return "=" + id;
+  }
+
+  private void insertProperty(String key, @Nullable Integer resourceId, @Nullable Integer userId) {
+    db.executeUpdateSql(format(
+            "insert into properties (prop_key,resource_id,text_value,user_id) values ('%s',%s,'%s',%s)",
```
Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java]: indentation seems bad here

**Comment by julienlancelot** [comment]
Minor feedback to apply, then LGTM

File: sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java (hunk 4 above)
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java]: fixed

File: sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java (hunk 5 above)
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v54/RemoveComponentPagePropertiesTest.java]: fixed

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Protected method 'configureModule(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/9ead984a861a6452e97f541909c0daf7e82509a9/sonar-db/src/main/java/org/sonar/db/version/MigrationStepModule.java#L68) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

**Comment by sns-seb** [comment]
merged

**Pull request closed without merging**